[ADD] pay_pro: add counterpart_currency

jjscarafia opened the pull request:
Lo traemos de acá https://github.com/ingadhoc/account-payment/tree/16.0/account_payment_group_currency

jjscarafia committed:
[ADD] pay_pro: add counterpart_currency

Lo traemos de acá https://github.com/ingadhoc/account-payment/tree/16.0/account_payment_group_currency

roboadhoc commented:
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/account-payment/pull/591).

jjscarafia committed:
[REM] label_journal_id: not used anymore

jjscarafia committed:
[FIX] pay_pro: use_pay_pro computed again

La idea de que es computado es para que solo este disponible si se usa un metodo de pago que hace apuntes y la cia tiene marcado pay pro. Porque si el método no hace apuntes, en realidad los features de pay pro no funcionan (al menos por ahora, y tampoco nos interesa ese caso de uso por ahora)

jjscarafia committed:
[FIX] counterpart currency

Al final no podemos usar  _synchronize_to_moves porque en la generación original del asiento odoo usa _generate_journal_entry

Lo mas prolijo es ir por _prepare_move_line_default_vals

por ahora hicimos un parche en paralelo de l10n_ar_tax para que haya compatibilidad

mas adelante podríamos ver en realidad de que l10n_ar_tax no tenga que parchera _prepare_move_line_default_vals y que direcatamente acá usemos payment_total

jjscarafia committed:
[IMP] pay_pro: countepart payment

Use rate as main counteapart field instead of the amount.

esto da varias mejoras de usabilidad. Entre ellas que si por alguna razón se ponía importe en 0 el rate lo perdes.

jjscarafia committed:
[FIX] pay_pro: avoid recomutation of withholding taxes due to inverse
